EGO MAUL Enhancement.user.js

```diff
@@ -0,0 +1,265 @@
+    id_div.style.display = "flex";
+    id_div.style.fledDirection = "row";
+    id_div.style.paddingTop = "10px";
+    id_group.appendChild(id_div);
+    id_div.appendChild(createLinkButton("Steam", "https://steamcommunity.com/profiles/" + id, "_blank"));
+    id_div.appendChild(createLinkButton("GameME", "https://edgegamers.gameme.com/search?si=uniqueid&rc=all&q=" + SteamIDConverter.toSteamID(id), "_blank"));
+    id_div.appendChild(createLinkButton("SteamID (IO)", "https://steamid.io/lookup/" + id, "_blank"));
+    id_div.appendChild(createLinkButton("SteamID (UK)", "https://steamid.uk/profile/" + id, "_blank"));
+
+    // IP buttons
+    var ip_group = Array.from(document.querySelectorAll(".control-label")).find(el => el.textContent === "IP").parentElement; // BECAUSE MAUL HAS THE IP LABELED WITH THE WRONG FOR
+    var ip = ip_group.querySelector("p").innerText;
+    var ip_div = document.createElement("div");
+    ip_div.style.display = "flex";
+    ip_div.style.fledDirection = "row";
+    ip_div.style.paddingTop = "10px";
+    ip_group.appendChild(ip_div);
+    ip_div.appendChild(createLinkButton("Check Spur", "https://spur.us/context/" + ip, "_blank"));
+    ip_div.appendChild(createLinkButton("Check IPInfo", "https://ipinfo.io/" + ip, "_blank"));
+}
+
+/**
+ * Automatically converts old links to updated ones
+ */
+function handleProfile() {
+    var userNotes = [...document.querySelectorAll("div.col-xs-6 > div > div:nth-child(3)")];
+    userNotes.forEach(userNote => {
+        userNote.textContent = userNote.textContent.replaceAll(
+            /(?:https?:\/\/)?(?:www\.)?edge-gamers\.com\/forums\/showthread\.php\?p=(\d+)(?:#?post(\d+))?/g,
+            function (match, threadId, postId) {
+                return generateForumsURL(threadId, postId);
+            });
+        userNote.textContent = userNote.textContent.replaceAll(
+            /(?:https?:\/\/)?(?:www\.)?edge-gamers\.com\/forums\/showthread\.php\?(\d+)[\-a-zA-Z]*/g,
+            function (match, threadId) {
+                return generateForumsURL(threadId, null);
+            });
+    });
+}
+
+/**
+ * Updates banning admins and ban notes
+ */
+function handleBanList() {
+    convertBanningAdmins();
+    updateBanNoteURLs();
+}
+
+/**
+ * Adds hyperlinks to the Banning Admins fields
+ */
+function convertBanningAdmins() {
+    if (Object.keys(knownAdmins).length === 0)
+        loadAdmins();
+    let expandingElementList = document.querySelectorAll(".expand");
+    for (let expandingElement of expandingElementList) {
+        let element = expandingElement.childNodes[1];
+        let adminsOrReason = element.childNodes[14].textContent;
+        let adminsElement;
+        if (adminsOrReason === "Admins Online:") {
+            // Admins Online field, fetch 17th
+            adminsElement = element.childNodes[17];
+        } else {
+            // Reason field is given, fetch 24th
+            adminsElement = element.childNodes[24];
```

Comment:
Could we modify this to use querySelector as opposed to indexes of childNodes? Using indexes of childNodes can easily result in issues down the line if MAUL updates the format, or even if another script inserts something into the element.